res/res_http_websocket.c

```diff
@@ -1441,6 +1441,10 @@ struct ast_websocket *AST_OPTIONAL_API_NAME(ast_websocket_client_create)
 	(const char *uri, const char *protocols, struct ast_tls_config *tls_cfg,
 	 enum ast_websocket_result *result)
 {
+	if (!uri) {
+		return NULL;
+	}
+
 	struct ast_websocket_client_options options = {
 		.uri = uri,
 		.protocols = protocols,
```

Comment:
Besides the other comments, you can't put code before declarations...
res_http_websocket.c: In function '__ast_websocket_client_create':
res_http_websocket.c:1448:2: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
  struct ast_websocket_client_options options = {
  ^~~~~~